### DISCUSSION TITLE
Can you help with compiling `libtins` library using CPM?

### DISCUSSION THREAD

**Discussion by traversebitree** [cpm-cmake/CPM.cmake#545] [Q&A]
I'm trying to use CPM to compile the `libtins` library as a dependency for my application. The following is what I wrote, but it keeps failing, the OS is Windows 11.

```cmake
CPMAddPackage(
    NAME winflexbison
    GIT_TAG v2.5.25
    GITHUB_REPOSITORY lexxmark/winflexbison
    SYSTEM
    OPTIONS "BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS}"
)

CPMAddPackage(
    NAME libpcap
    GIT_TAG libpcap-1.10.4
    GITHUB_REPOSITORY the-tcpdump-group/libpcap
    SYSTEM
    OPTIONS "BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS}"
            "DISABLE_BLUETOOTH TRUE"
            "DISABLE_DBUS TRUE"
            "DISABLE_DPDK TRUE"
            "DISABLE_RDMA TRUE"
            "DISABLE_DAG TRUE"
            "DISABLE_SEPTEL TRUE"
            "DISABLE_TC TRUE"
            "ENABLE_REMOTE FALSE"
)

CPMAddPackage(
    NAME libtins
    GIT_TAG v4.5
    GITHUB_REPOSITORY mfontanini/libtins
    SYSTEM
    OPTIONS "BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS}"
            "LIBTINS_ENABLE_CXX11 TRUE"
            "LIBTINS_ENABLE_ACK_TRACKER FALSE"
            "LIBTINS_ENABLE_WPA2 FALSE"
            "LIBTINS_ENABLE_DOT11 FALSE"
)
```

![image](https://github.com/cpm-cmake/CPM.cmake/assets/30373236/cafdcc4b-ea6f-40d7-9ab6-698f522be026)

**Comment by Avus-c** [comment]
The error is generated from a call to `find_program` (`find_program(LEX_EXECUTABLE NAMES flex win_flex lex)`) in libpcap's CMakeLists.txt. `libpcap` expects a win_flex executable to be present during the configuration.

You can build winflex separately before proceeding with your project.

Alternatively, you can add an `execute_process` instruction between winflex and libpcap to build winflex before cmake parses the libpcap CMakeLists.txt. It should look something like this:

```cmake
CPMAddPackage(
    NAME winflexbison
    ...
)

execute_process( <configure winflex project> )
execute_process( <build_command> )

CPMAddPackage(
    NAME libpcap
   ...
)
```

In either case, you may have to tell cmake where it can find the executable.

This may further assist you or guide you on the right track: [StackOverflow link](https://stackoverflow.com/questions/37553280/how-to-build-cmake-externalproject-while-configurating-main-one) (it uses ExternalProject_Add with a helper subproject, but essentially is the same thing).

Hope I could help at least a little. 😄
